packages/kbn-palettes/classes/color_fn_palette.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { Optional } from 'utility-types';
+import { KbnBasePaletteConfig, KbnBasePalette } from './palette';
+import { IKbnPalette, KbnPaletteType } from './types';
+
+const DEFAULT_COLOR_COUNT = 10;
+
+export interface KbnColorFnPaletteConfig extends Optional<KbnBasePaletteConfig, 'colorCount'> {
+  type: KbnPaletteType;
+  colorFn: (n: number) => string[];
+  defaultNumberOfColors?: number;
+}
+
+export class KbnColorFnPalette extends KbnBasePalette implements IKbnPalette {
```

Comment:
Another naming question: why this is marked as `Fn` but the `KbnGradientPalette`, which implements the same type is not?

Reply:
Yeah I named it this was because this derives the palette from color function (i.e. `euiPaletteCool`) which can be either gradient or categorical but both result in a `IKbnPalette`.